### PR TITLE
Data Explorer Updates

### PR DESCRIPTION
**Pull request by eberger727**
Data Explorer has an updated dashboard that uses FireFly's [new charting capabilities](https://github.com/hyperledger/firefly/pull/341), along with new Operations, Events, and DataType pages:

## Dashboard
Dashboard

## Message Details
MessageDetails

## Operations
Operations

## Events
Events

## Video

https://user-images.githubusercontent.com/17093033/144930425-df18923a-bf1d-4e89-9f42-f7f0d8f2bbba.mov

### CODE REVIEW

File: src/modules/data/views/Messages/Messages.tsx [hunk 1]
```diff
@@ -14,31 +14,28 @@
...
+import { FilterModal } from '../../../../core/components/FilterModal';
+import { ApplicationContext } from '../../../../core/contexts/ApplicationContext';
+import { IHistory, IMessage } from '../../../../core/interfaces';
 import { filterOperators } from '../../../../core/utils';
-import { DatePicker } from '../../../../core/components/DatePicker';
+import { useDataTranslation } from '../../registration';
+import { MessageList } from './MessageList';
+import { MessageTimeline } from './MessageTimeline';
 
 export const Messages: () => JSX.Element = () => {
```
Comment (shorsher) [inline src/modules/data/views/Messages/Messages.tsx:33]: Does the histogram endpoint support messages? Hoping we could have those awesome charts in the messages view.

File: src/modules/data/views/Messages/MessageDetails.tsx [hunk 2]
```diff
@@ -13,146 +13,51 @@
...
-        <CopyToClipboard text={hash}>
-          <Button size="small" className={classes.copyButton}>
-            {t('copy')}
-          </Button>
-        </CopyToClipboard>
-      </Grid>
-    </Grid>
-  );
-
-  const transactionLink = (txId: string | undefined) => (
```
Comment (shorsher) [inline src/modules/data/views/Messages/MessageDetails.tsx:120]: I think it's important for us to provide a link to the transaction the message was in, if there was one.

File: src/modules/data/views/Messages/MessageDetails.tsx [hunk 3]
```diff
@@ -13,146 +13,51 @@
...
 import { useDataTranslation } from '../../registration';
 
-interface Props {
-  message: IMessage;
-  open: boolean;
-  onClose: () => void;
-}
-
-export const MessageDetails: React.FC<Props> = ({ message, open, onClose }) => {
+export const MessageDetails: () => JSX.Element = () => {
```
Comment (shorsher) [inline src/modules/data/views/Messages/MessageDetails.tsx:36]: Seeing a few responsiveness issues on smaller screen widths
Screen Shot 2021-12-09 at 10 11 36 AM

**Review by shorsher — COMMENTED** [review]
Just a few comments here, also pointing out the discussion we had around chart responsiveness on the dashboard. Moving 
from 1x4 to 2x2 & 4x1 depending on screen widths. 

edit: Forgot to mention, in addition to messages, can we add the histogram to the transactions view?


These updates are really great, thanks @eberger727!

File: src/modules/data/views/Dashboard.tsx [hunk 4]
```diff
@@ -14,133 +14,393 @@
...
+            </>
+          );
+        })}
+      </Grid>
+    </Paper>
+  );
+
+  const summaryPanel = (data: ISummaryPanel): JSX.Element => (
+    <Card sx={{ height: '140px', width: '100%' }}>
+      <CardContent className={classes.content}>
```
[the remark this answers is not in the document]
Comment (eberger727) [reply src/modules/data/views/Dashboard.tsx:371]: Maybe both? I like the `sx` pattern for tweaking inline things here and there, but for larger components it's probably best to use `makeStyles`

File: src/modules/data/views/Messages/Messages.tsx (hunk 1 above)
Comment (eberger727) [reply src/modules/data/views/Messages/Messages.tsx:33]: Good idea, added it for messages and transactions list views

Comment (eberger727) [reply src/modules/data/views/Messages/Messages.tsx:33]: ### Transactions
tx

### Messages
msgs

File: src/modules/data/views/Dashboard.tsx [hunk 5]
```diff
@@ -14,133 +14,393 @@
...
+}
+
+interface ISummaryPanel {
+  metrics: IMetric[];
+  title: string;
+  total: string | undefined;
+}
+
+const MSG_STATUS_COLORS: { [key in MSGStatus]: string } = {
+  confirmed: '#462DE0',
```
[the remark this answers is not in the document]
Comment (eberger727) [reply src/modules/data/views/Dashboard.tsx:63]: added an FFColors enum to interface files

File: src/modules/data/views/Messages/MessageDetails.tsx (hunk 3 above)
Comment (eberger727) [reply src/modules/data/views/Messages/MessageDetails.tsx:36]: Fixed:
fixed

File: src/modules/data/views/Messages/MessageDetails.tsx (hunk 2 above)
Comment (eberger727) [reply src/modules/data/views/Messages/MessageDetails.tsx:120]: Added:
image

Comment (shorsher) [reply src/modules/data/views/Messages/MessageDetails.tsx:120]: On this one, can we also show the transaction hash (if there is one) in a popover?

Comment (eberger727) [reply src/modules/data/views/Messages/MessageDetails.tsx:120]: https://user-images.githubusercontent.com/17093033/145601385-dee35e46-e596-4ec9-aa58-7683a7ba9886.mov

Comment (eberger727) [reply src/modules/data/views/Messages/MessageDetails.tsx:120]: https://user-images.githubusercontent.com/17093033/145601508-c94359ac-a62a-4a3e-b686-719e39ac4354.mov

**Comment by shorsher** [comment]
LGTM to me 🚀 

Awesome work

**Review by shorsher — APPROVED** [review]

**Pull request merged**